### PR TITLE
Add technofundamental demo

### PR DESCRIPTION
**Pull request by psonice**
Included the technofundamental demo (requires an iPhone X or later)
Also made a couple of changes to Demo and ViewController to support demo requirements, and pre-filtering the demo list to hide unsupported demos. I've only added metal_v4 as a requirement but it should be easily extended if needed.

### CODE REVIEW

File: jmlcollection/ViewController.swift [hunk 1]
```diff
@@ -22,33 +22,36 @@ import UIKit
...
-        Demo(name: "Worlds", group: "Dekadence", year: "2017", viewController: { return WorldsViewController() }, htmlFilename: nil),
-        Demo(name: "#jumalauta", group: "Jumalauta", year: "2017", viewController: nil, htmlFilename: "jml_irc_ad.html"),
-        Demo(name: "Honey And Whore", group: "Jumalauta", year: "2017", viewController: nil, htmlFilename: "jml17.html"),
-        Demo(name: "Δ", group: "Jumalauta", year: "2017", viewController: nil, htmlFilename: "jmldeltaindex.html"),
-        Demo(name: "Thinkpiece", group: "Dekadence", year: "2017", viewController: nil, htmlFilename: "thinkpiece.html"),
-        Demo(name: "Party Hard 2", group: "Jumalauta", year: "2016", viewController: nil, htmlFilename: "jmlpartyhard2index.html"),
-        Demo(name: "thoron is not the answer to every quiplash question", group: "Jumalauta", year: "2016", viewController: nil, htmlFilename: "halfhourshit.html"),
-        Demo(name: "Destroy All Humans", group: "Jumalauta", year: "2015", viewController: nil, htmlFilename: "jmldstr.html"),
-    ]
+	var demos: [Demo] = [
```
Comment (superjohan) [inline jmlcollection/ViewController.swift]: Please use spaces for indentation. :)

File: jmlcollection/ViewController.swift [hunk 2]
```diff
@@ -22,33 +22,36 @@ import UIKit
...
+		Demo(name: "technofundamental", group: "CRTC + RiFT", year: "2018", requirements: .metal_v4, viewController: { return TechnofundamentalViewController() }, htmlFilename: nil),
```
Comment (superjohan) [inline jmlcollection/ViewController.swift]: Try keeping this in chronological order, at least for now. We could add sorting options / filtering at some later point I think.

File: jmlcollection/ViewController.swift (hunk 1 above)
Comment (psonice) [reply jmlcollection/ViewController.swift]: Lol, didn't even notice :D (At least it didn't update every last file with tabs...)

File: jmlcollection/ViewController.swift (hunk 2 above)
Comment (psonice) [reply jmlcollection/ViewController.swift]: I wondered about that - the demo was last year, but the port is fresh (and actually has some improvements over the desktop version) so I left it there. I'll leave it to your judgement :)

**Comment by superjohan** [comment]
Yeah, that's good. I'll make sure everything works later. You ok with me releasing this after Revision sometime? I'd like to include my Revision demo in the next release.

**Comment by psonice** [comment]
Oh, cool news on the revision demo! Yeah, that's totally fine by me. 

Suggestion: if your revision demo is done early (and not 1 hour after the compo deadline :D) you can submit it to Apple before revision, and set it to developer release. Then it goes through app review, and just sits there ready to go. You can push it to the store during the compo if you want.

**Comment by superjohan** [comment]
That's what I did last year. :) The problem is that the compo version is very likely going to be a bit unfinished, in that it currently only properly supports the iPhone X resolution. If I have the time I'll submit it it before Revision, but it's quite doubtful.

**Comment by superjohan** [comment]
One thing that would help me is prefixing classes and such, since there may be some overlap in the future. (at least with generic names such as `Demo.metal` etc) One day I'll learn how to use modules. :)

**Pull request merged**